devito/data/data.py

```diff
@@ -210,9 +210,21 @@ def _normalize_index(self, idx):
             if any(i is Ellipsis for i in idx):
                 # Explicitly replace the Ellipsis
                 items = (slice(None),)*(self.ndim - len(idx) + 1)
-                return idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
+                items = idx[:idx.index(Ellipsis)] + items + idx[idx.index(Ellipsis)+1:]
             else:
-                return idx + (slice(None),)*(self.ndim - len(idx))
+                items = idx + (slice(None),)*(self.ndim - len(idx))
+            # Normalize slice steps:
+            processed = []
```

Comment:
this whole loop looks like is expressible as
```
processed = [slice(i.start, i.stop, 1) (if isinstance(i, slice) and i.step is None) else i
             for i in items]
```

Reply:
Updated.

Reply:
Feels like my previous comment is obsolete and should now be ignored ?